src/Orleans/Streams/Predicates/IStreamNamespacePredicate.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Orleans.Streams
+{
+    public interface IStreamNamespacePredicate
+    {
+        bool IsMatch(string streamNamespace);
```

Comment:
I think we should be careful here to introduce filtering on streamId  into implicit subscribe. Aside from the performance penalty DixonDs brought up, it conflicts with the nature of explicit subscribe too. Because of the dynamic nature of streamId,  filtering/subscribe to a stream fits more naturally into the territory of explicit subscribe. Adding the same feature into implicit subscribe may bring more confusion to users , potentially bugs in edge cases. I think we should defer this feature until specifically asked by users